code/modules/antagonists/heretic/magic/shadow_cloak.dm

```diff
@@ -28,7 +28,10 @@
 	return ..()
 
 /datum/action/cooldown/spell/shadow_cloak/is_valid_target(atom/cast_on)
-	return isliving(cast_on) && !HAS_TRAIT(cast_on, TRAIT_HULK) // Hulks are not stealthy. Need not apply
+	if(HAS_TRAIT(cast_on, TRAIT_HULK))
+		cast_on.balloon_alert(cast_on, "cannot cast while hulk")
```

Comment:
this is a proc to check whether or not tis a valid target, it should not apply the balloon itself i think?

Reply:
returning false will instantly cancel the spell chain iirc

Reply:
I'm not at my desktop or I could look at the actual call chain